server/db/db.js

```diff
@@ -636,6 +637,34 @@ async function getInstallationWithSessionId(sessionId, clientParam) {
   return null
 }
 
+async function getHistoricAlertsByAlertApiKey(alertApiKey, maxHistoricAlerts, maxTimeAgoInMillis, clientParam) {
+  let client = clientParam
+  const transactionMode = typeof client !== 'undefined'
+
+  try {
+    if (!transactionMode) {
+      client = await pool.connect()
+    }
+
+    // prettier-ignore
+    const { rows } = await client.query('SELECT s.id, b.unit, s.incident_type, s.num_presses, s.created_at, s.responded_at FROM sessions AS s LEFT JOIN buttons AS b ON s.button_id = b.button_id LEFT JOIN installations AS i ON s.installation_id = i.id WHERE i.alert_api_key = $1 AND (s.state = $2 OR s.updated_at < now() - $3 * INTERVAL \'1 millisecond\') ORDER BY s.created_at DESC LIMIT $4', [alertApiKey, ALERT_STATE.COMPLETED, maxTimeAgoInMillis, maxHistoricAlerts])
```

Comment:
I think this is our most complicated SQL query so far in this code base. But don't worry, I wrote integration tests for it. Let me know what you think.

Reply:
@trisapeace that would be awesome, thanks.

Reply:
Done.